src/realm/object-store/c_api/app.cpp

```diff
@@ -948,7 +948,7 @@ RLM_API bool realm_mongo_collection_find(realm_mongodb_collection_t* collection,
     return wrap_err([&] {
         collection->find_bson(convert_to_bson<bson::BsonDocument>(filter_ejson),
                               to_mongodb_collection_find_options(options),
-                              [&](util::Optional<bson::Bson> bson, util::Optional<AppError> app_error) {
+                              [=](util::Optional<bson::Bson> bson, util::Optional<AppError> app_error) {
```

Comment:
Only a minimal thing, I think the only thing you need to capture by copy is the `callback`, the rest of the data can be passed by reference. Although in 99% of the case, we are copying a `ptr` (thus equivalent to passing the arguments by reference), for `realm_string_t filter_ejson` we will be copying `sizeof(ptr) + sizeof(size_t)` bytes. 
Not a big issue, since the amount of data copied will always be constant.

Reply:
I need to pass by value `data`, `delete_data` and `callback`. Otherwise when I receive the callback I can't access the data object.